src/renderer/components/CaptionProgram.tsx

```diff
@@ -0,0 +1,336 @@
+  return () => { hasStopped = true; };
+};
+
+
+const startShowingText = function(el : HTMLElement, hastebinId : string) {
+  if (hastebinId === 'test') {
+    let testProgram = `
+    setBlinkDuration 300
+    setBlinkDelay 100
+    setBlinkGroupDelay 1200
+    setCaptionDuration 2000
+    setCaptionDelay 1200
+
+    bigcap YOU LOVE FLUFFY KITTENS
+    blink KITTENS / ARE / YOUR / LIFE
+    cap Cuddle all the kittens forever because you love them.
+    `;
+    return startText(el, testProgram);
+  }
+
+  // const url = `text/${pastebinId}.txt`;
+  // const url = 'https://crossorigin.me/http://pastebin.com/raw/' + pastebinId;
+  // const url = 'http://cors-proxy.htmldriven.com/?url=http://pastebin.com/raw/' + hastebinId;
+  // const url = 'http://cors-proxy.htmldriven.com/?url=https://hastebin.com/raw/' + hastebinId;
+  // const url = 'https://cors-anywhere.herokuapp.com/https://hastebin.com/raw/' + hastebinId;
+  const url = 'https://hastebin.com/raw/' + hastebinId;
+
+  let _hasStoppedEarly = false;
+  let _stop = () => {
+    _hasStoppedEarly = true;
+  };
+  const stop = () => { _stop(); };
+  $.ajax({
+    url: url,
+    method: 'get',
+    success: function(data) {
+      if (_hasStoppedEarly) return;
+      try {
+        JSON.parse(data.body);
+        return;
+      } catch (e) {
+        // ok; old format
+      }
+      if (data.body) {
+        data = data.body;
+      }
+      console.log(data);
+      _stop = startText(el, data);
+    },
+    error: function(err) {
+      console.error("Could not load hastebin ID", hastebinId);
+      alert("Could not load your hastebin. This may or may not be temporary.");
+    },
+  });
+  return stop;
+};
+
+export default class CaptionProgram extends React.Component {
+  readonly el = React.createRef<HTMLDivElement>();
+
+  readonly props: {
+    hastebinID: string,
+  };
+
+  readonly state = {
+    stopFunc: Function(),
+    lastHastebinID: ""
+  };
+
+  shouldComponentUpdate(nextProps : {hastebinID : string}, nextState: any) {
+    return nextProps.hastebinID !== this.props.hastebinID;
+  }
+
+  componentWillReceiveProps(nextProps : {hastebinID : string}) {
+    this._update(nextProps);
+  }
+
+  componentWillUnmount() {
+    PROGRAM = [];
+    programCounter = 0;
+    this.state.stopFunc();
+  }
+
+  _update(props: {hastebinID : string}) {
+    if (!this.el.current) return;
+    if (props.hastebinID == this.state.lastHastebinID) return;
+    this.setState({lastHastebinID: props.hastebinID});
+    this._stop();
+    this.setState({stopFunc: startShowingText(this.el.current, props.hastebinID)});
+  }
+
+  _stop() {
+    if (this.state.stopFunc) {
+      this.state.stopFunc();
+      this.setState({stopFunc: null});
+    }
+  }
+
+  render() {
+    return (
+      <div style={{pointerEvents: 'none', display: 'table', width: '100%', height: '100%'}}
+          className="u-fill-container text-display">
+        <div style={{display: 'table-cell'}} ref={this.el} />
+      </div>
+    );
+  }
+}
```

Comment:
Good work updating this for TypeScript. I'm mostly not reviewing it except to identify how to remove jQuery. :-)